docs/source/howtos/add_instrument.rst

```diff
@@ -0,0 +1,105 @@
+How To Add an Instrument
+========================
+
+ResINS strives to model all :term:`INS` :term:`instruments<instrument>` past,
+present, and future, but it is inevitable that some will not be natively
+supported *yet*. Fortunately, ResINS was designed to make adding new instruments
+relatively straightforward. Though, there are two ways of doing this:
+
+.. contents::
+    :backlinks: entry
+    :depth: 2
+    :local:
+
+.. _add-personal-instrument:
+
+How To Add an Instrument for Personal Use
+-----------------------------------------
+
+It is possible to add an :term:`instrument` only locally, outside the ResINS
+ecosystem. Technically, all that is required is a dictionary, but the most
+readable way is to construct a YAML file with the instrument details. This file
+**must** follow the :doc:`data file spec<../dev/yaml_spec>` and include all the
+necessary information. Then, ResINS can be leveraged to do the rest without
+additional code:
+
+>>> from resolution_functions import Instrument
+>>> new_instrument_path = '~/instrument/instrument.yaml'
+>>> version = 'version'  # If the created YAML file contains multiple versions
+>>> new_instrument = Instrument.from_file(new_instrument_path, version)
+>>> new_instrument.name
+'new_instrument'
+>>> new_instrument.version
+'version'
+
+Given that the YAML file specifies a model already implemented in ResINS, and
+that its parameters have been all included, the resolution function can be
+computed:
+
+>>> model = new_instrument.get_resolution_function('PyChop_fit', chopper_package='A', e_init=100, chopper_frequency=300)
+>>> print(model)
+PyChopModelFermi(citation=[''])
+
+However, if a new model was created for the new instrument, more work will be
+required, see :doc:`add_model`.
+
+How to register an instrument with ResINS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+??????????????????
```

Comment:
Do we want to have the ability (for people who are creating a new instrument without changing ResINS) to insert the new instrument file into `INSTRUMENT_MAP`? That way, they would be to use `Instrument.from_default` instead of `Instrument.from_file`. All we would need to do for this to be possible is to allow `INSTRUMENT_MAP` entries to also be full paths, which could be done by adding a `os.path.exists` check to `Instrument._get_file`.

Reply:
That seems pretty innocuous. Do you have use case in mind where it wouldn't be convenient to use `from_file`?

Reply:
Not in particular, but I suppose it would depend on the circumstances under which someone might want to create a new instrument without opening a PR for us. In a script or jupyter notebook, I don't think it would be difficult to keep a variable with the path to their YAML file around, but for anything more complicated (like software), I imagine it might be easier to register once and then simply call `from_default`.

I am not sure this is important to have, but I wanted to check in case it is.

Reply:
I would think that "more complex software" use-case is better served by a plugin system; that in turn would probably have an `add_to_instrument_map` function somewhere which can be used for hot-loading while developing said software.